EatGoodNaija.Server/Controllers/CartController.cs

```diff
@@ -0,0 +1,39 @@
+﻿using EatGoodNaija.Server.Model;
+using EatGoodNaija.Server.Services.Interface;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EatGoodNaija.Server.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CartController : ControllerBase
+    {
+        private readonly ICartRepository _cartRepository;
+
+        public CartController(ICartRepository cartRepository)
+        {
+            _cartRepository = cartRepository;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<CustomerCart>> GetCartById(string id)
+        {
+            var cart = await _cartRepository.GetCartAsync(id);
```

Comment:
Your controllers should be communicating with your services while your services communicate with the repository